### PR TITLE
API Calls

### PR DESCRIPTION
**Pull request by amosNash**
This pull request is for my API calls. I have also implemented test for my API calls using jest, and I also refactored test for my database using jest. The bulk of the changes are in the test folder and specifically api.test.js.

### CODE REVIEW

File: test/database.test.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const item = new Description({productId: 900});
+    item.save()
+      .then(() => {
+        assert(!item.isNew);
+        done();
+      });
+  });
+
+
+});
```
Comment (jrozal) [inline test/database.test.js:66]: Good test coverage for database!

File: server/index.js [hunk 2]
```diff
@@ -1,18 +1,48 @@
...
+      next();
+    }
+  });
+});
 
-app.get('/description/1000', (req, res) => {
-  var data = db.findOne({productId: 1000});
-  res.send(data);
+//get multiple item descriptions
+app.post('/description/multiple', (req, res, next) => {
```
Comment (jrozal) [inline server/index.js]: Is there a reason for using a post request here? The function looks like it is querying and reading data, as opposed to inserting data.

File: test/api.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    test('/description/multiple endpoint should send back multiple descriptions', async (done) => {
+      const res = await request.post('/description/multiple').send(productIds);
+      expect(productIds.length).toBe(3);
+      done();
+    })
+  })
+})
+
+
```
Comment (jrozal) [inline test/api.test.js:43]: I think you did a good job testing all the various api endpoint responses and including tests for both 200 and 404 responses.

File: test/api.test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      expect(res.body.length).toBe(1);
+      done();
+    })
+  })
+
+  describe('/description/multiple Endpoint', () => {
+    const productIds = [1001, 1002, 1003];
+
+    test('/description/multiple endpoint will respond with 200 if array of productIds between 1000-1090 are sent', async (done) => {
+      const res = await request.post('/description/multiple').send({productIds: productIds});
```
Comment (jrozal) [inline test/api.test.js]: Same question as I have on line 34 of your `server/index.js` file.

File: database/query.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+
+
+
+module.exports = {
+  getDescriptionForOneProduct: getDescriptionForOneProduct,
+  getDescriptionForMultipleProducts: getDescriptionForMultipleProducts,
+  deleteProduct: deleteProduct
+};
```
Comment (jrozal) [inline database/query.js:42]: Straight forward, easy to read database functions. All looks good here!

Comment (jrozal) [reply database/query.js:42]: I also like that you separated your query functions from the server file. Nice organization/file structure!

**Review by jrozal — COMMENTED** [review]
Hi Amos,
Your test coverage looks good! I also appreciate the cleanup of comments/code that you did. I left a couple comments regarding your axios calls and whether a different http method should be used. [Get vs Post](https://www.w3schools.com/tags/ref_httpmethods.asp)

File: server/index.js (hunk 2 above)
Comment (amosNash) [reply server/index.js]: I think you're right that a "get" request makes more sense here. I'm going to go ahead and make the change.

File: test/api.test.js (hunk 4 above)
Comment (amosNash) [reply test/api.test.js]: I'm going to go ahead and change it to a "get" request as it is not inserting any data into my database.

**Review by jrozal — COMMENTED** [review]
Changes look good 👍

**Pull request merged**